components/dashboard/src/admin/BlockedRepositories.tsx

```diff
@@ -0,0 +1,275 @@
+                                    fill="#A8A29E"
+                                />
+                            </svg>
+                        </div>
+                        <input
+                            type="search"
+                            placeholder="Search by URL Regex"
+                            onKeyDown={(ke) => ke.key === "Enter" && search()}
+                            onChange={(v) => {
+                                setQueryTerm(v.target.value.trim());
+                            }}
+                        />
+                    </div>
+                    <div className="flex space-x-2">
+                        <button disabled={searching} onClick={search}>
+                            Search
+                        </button>
+                        <button onClick={add}>New Variable</button>
+                    </div>
+                </div>
+            </div>
+            <div className="flex flex-col space-y-2">
+                <div className="px-6 py-3 flex justify-between text-sm text-gray-400 border-t border-b border-gray-200 dark:border-gray-800 mb-2">
+                    <div className="w-8">ID</div>
+                    <div className="w-9/12">URL Regex</div>
+                    <div className="w-1/12">Block user</div>
+                    <div className="w-2/12">Delete</div>
+                </div>
+                {searchResult.rows.map((br) => (
+                    <BlockedRepositoryEntry br={br} confirmedDelete={confirmDeleteBlockedRepository} />
+                ))}
+            </div>
+        </>
+    );
+}
+
+function BlockedRepositoryEntry(props: { br: BlockedRepository; confirmedDelete: (br: BlockedRepository) => void }) {
+    return (
+        <div className="rounded-xl whitespace-nowrap flex py-6 px-6 w-full justify-between hover:bg-gray-100 dark:hover:bg-gray-800 focus:bg-gitpod-kumquat-light group">
+            <div className="flex flex-col w-8 truncate">
+                <span className="mr-3 text-lg text-gray-600 truncate">{props.br.id}</span>
+            </div>
+            <div className="flex flex-col w-9/12 truncate">
+                <span className="mr-3 text-lg text-gray-600 truncate">{props.br.urlRegexp}</span>
+            </div>
+            <div className="flex flex-col self-center w-1/12">
+                <CheckBox title={""} desc={""} checked={props.br.blockUser} disabled={true} />
+            </div>
+            <div className="flex flex-col w-2/12">
+                <button onClick={() => props.confirmedDelete(props.br)}>Delete</button>
+            </div>
+        </div>
+    );
+}
+
+interface AddBlockedRepositoryModalProps {
+    blockedRepository: NewBlockedRepository;
+    onClose: () => void;
+    save: (br: NewBlockedRepository) => void;
+}
+
+function AddBlockedRepositoryModal(p: AddBlockedRepositoryModalProps) {
+    const [br, setBr] = useState({ ...p.blockedRepository });
+    const ref = useRef(br);
```

Comment:
How come we need the `useRef` hook here? It' not something I've seen used much before?

Reply:
Because we don't have access to the state directly here, but don't want to manipulate the part of the state that is displayed, but just a version of it.
Maybe this can solves differently and easier. This is what I saw in the EnvVarSettings. :+1: